stdlib/public/core/StringUTF8.swift

```diff
@@ -174,10 +174,10 @@ extension String {
 
     init(_ _core: _StringCore) {
       self._core = _core
-      self._endIndex = Index(_core, _core.endIndex, Index._emptyBuffer)
+      self._endIndex = Index(_core.endIndex, Index._emptyBuffer)
```

Comment:
Slicing a `UTFView` uses a start and end index that are distinct from the start and end index of the underlying core. To drop `_endIndex` we'd need to actually slice the string core and I think we'd cause a regression in index sharing (which currently works for `UTF8View`).